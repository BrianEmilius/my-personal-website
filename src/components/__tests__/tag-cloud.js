import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import TagCloud from "../tag-cloud";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      allMarkdownRemark: {
        edges: [
          {
            node: {
              frontmatter: {
                tags: ["ost", "feta"]
              }
            }
          },
          {
            node: {
              frontmatter: {
                tags: ["gouda"]
              }
            }
          }
        ]
      }
    })
  );
});

describe("TagCloud", function() {
  it("renders correctly", function() {
    const tree = renderer.create(<TagCloud />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
