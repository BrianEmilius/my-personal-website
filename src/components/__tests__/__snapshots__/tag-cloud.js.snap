// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCloud renders correctly 1`] = `
<div
  className="tagContainer"
>
  <a
    className="tag"
    href="/tags/ost"
    onClick={[Function]}
  >
    #
    ost
  </a>
  <a
    className="tag"
    href="/tags/feta"
    onClick={[Function]}
  >
    #
    feta
  </a>
  <a
    className="tag"
    href="/tags/gouda"
    onClick={[Function]}
  >
    #
    gouda
  </a>
</div>
`;
